apis/python/src/tiledb/vector_search/ingestion.py

```diff
@@ -2505,15 +2505,20 @@ def consolidate_and_vacuum(
         config: Optional[Mapping[str, Any]] = None,
     ):
         with tiledb.Group(index_group_uri) as group:
+            write_group = tiledb.Group(index_group_uri, "w")
```

Comment:
There are multiple Group open operations in `consolidate_and_vacuum` as well as before calling it, can we open the group once in `m` mode?

Reply:
Updated to just have a single `w` within `consolidate_and_vacuum()`, and in a follow-up will look at if we can pass an open group to this method / have less opens in general.

Reply:
Hmm, it seems like my update to use a single write group in consolidate_and_vacuum() doesn't work in the cloud tests (https://github.com/TileDB-Inc/TileDB-Vector-Search/actions/runs/8848628336/job/24300692458?pr=348#step:6:378):
```
https://github.com/TileDB-Inc/TileDB-Vector-Search/actions/runs/8848628336/job/24298822778?pr=348#step:6:483 
=========================== short test summary info ============================
FAILED test/test_cloud.py::CloudTests::test_cloud_ivf_flat - tiledb.cc.TileDBError: [TileDB::REST] Error: Error in libcurl PATCH operation: libcurl error message 'CURLE_OK'; HTTP code 500; server response data '{"code":901,"message":"failed to write member changes to core group: asset not found as an array or group: record not found","request_id":"b6c4b094-3755-40e7-9ea2-727d297b19dc"}'.
FAILED test/test_cloud.py::CloudTests::test_cloud_ivf_flat_random_sampling - tiledb.cc.TileDBError: [TileDB::REST] Error: Error in libcurl PATCH operation: libcurl error message 'CURLE_OK'; HTTP code 500; server response data '{"code":901,"message":"failed to write member changes to core group: asset not found as an array or group: record not found","request_id":"634558b7-2c7c-46a9-b1d8-000592eb12fb"}'.
FAILED test/test_cloud.py::CloudTests::test_cloud_vamana - tiledb.cc.TileDBError: [TileDB::REST] Error: Error in libcurl PATCH operation: libcurl error message 'CURLE_OK'; HTTP code 500; server response data '{"code":901,"message":"failed to write member changes to core group: asset not found as an array or group: record not found","request_id":"f3c8dc93-ce04-4129-91cc-a7c107b0f841"}'.
=========== 3 failed, 62 passed, 218 warnings in 1737.60s (0:28:57) ============
Error: Process completed with exit code 1.
```

Reply:
Talked offline, and agreed to go back to the original approach and consolidate group opens in a follow-up.